### DISCUSSION TITLE
Enhancement: Spinning ball or progress output of o1 "thinking" process

### DISCUSSION THREAD

**Discussion by leondape** [danny-avila/LibreChat#4241] [Feature Requests & Suggestions]
### What features would you like to see added?

Probably this is obvious but I couldn't find an issue about this: 
When using o1, we don't really have any hints about the "thinking" process. ChatGPT even outputs what it is thinking about. Don't know if that is possible over the API. 
If it is not, I suggest having at least some form of "Thinking..." Animation that shows more activity than the current "pulsing-icon" animation. 
Pulsing icon on very processing time could be mistaken as a but that is easy to be mistaken as a bug. 
Having a dedicated Thinking animation would make it a little clearer to users. 

Surely, best Case would be to see the thinking steps as in ChatGPT.

### More details

nothing.

### Which components are impacted by your request?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Not a bad idea but I wonder if OpenAI would ever release the "chain of thought" in order to display that instead. They intentionally left it out of the API for "competitive reasons"

**Reply by thefunny1983** [reply]
Well the "competitive reasons" seem to have disappeared now with DeepSeek. ChatGPT is showing chain of thought now.

**Comment by bendavis78** [comment]
Has anyone inspected the output of o1 on the /completions API to see if it provides such info mid-stream?

**Reply by RedwindA** [reply]
> Has anyone inspected the output of o1 on the /completions API to see if it provides such info mid-stream?

there's no need to inspect it. openai said it wont be provided
![image](https://github.com/user-attachments/assets/a4f1aad9-ff42-4742-b02d-e0f2dc390f70)

**Reply by danny-avila** [reply]
It does not provide info mid-stream. I remember reading this was for competitive reasons. The fact that they released streaming without any intermediary outputs doesn't get my hopes up, though, but who knows if they will release it.

**Comment by DovydasNavickas** [comment]
The point about "Thinking..." indication still stands. It's very unclear whether something went wrong or it's still "thinking". Can you point to where this should be added? I could make a PR.

**Reply by leondape** [reply]
I believe it works now wherever possible. 
OpenAI API does not stream the thinking Process, so there it is not available. 
The other LLMs have it integrated I think
